lua/nvim-lsp-ts-utils/inlay-hints.lua

```diff
@@ -2,99 +2,156 @@ local o = require("nvim-lsp-ts-utils.options")
 local u = require("nvim-lsp-ts-utils.utils")
 
 local api = vim.api
-
+local INLAY_HINTS_METHOD = "typescript/inlayHints"
 local M = {}
 
-M.state = {
-    enabled = false,
-    ns = api.nvim_create_namespace("ts-inlay-hints"),
-}
-
-function M.setup_autocommands()
-    u.buf_autocmd(
-        "TSLspImportOnCompletion",
-        "BufEnter,BufWinEnter,TabEnter,BufWritePost,TextChanged,TextChangedI",
-        "autocmd_fun()"
-    )
+local ns = api.nvim_create_namespace("ts-inlay-hints")
+M.ns = ns
+
+-- enabled[bufnr] = false
+M.enabled = {}
+
+local function buf_enabled(bufnr)
+    if bufnr ~= nil then
+        if bufnr == 0 then
+            bufnr = api.nvim_get_current_buf()
+        end
+        if M.enabled[bufnr] == nil then
+            M.enabled[bufnr] = false
+        end
+        return M.enabled[bufnr]
```

Comment:
If we strictly want to return a boolean (which I don't think is necessary) couldn't we just return `M.enabled[bufnr] ~= nil`?